src/services/googleDrive.service.ts

```diff
@@ -63,4 +62,38 @@ export const uploadSignatureImage = async (file: File, bookingDate: string) => {
     console.error("Error retrieving data", error);
     return undefined;
   }
-}
\ No newline at end of file
+}
+
+export const uploadPDFDocument = async (file: File, data: any) => {
+  try {
+    // Create a FormData object
+    const formData = new FormData();
+    const {Date, Boat, Type, Amount} = data
+    // Ensure the file is a PDF
+    if (file.type !== "application/pdf") {
+      throw new Error("Only PDF files are allowed");
+    }
+    formData.append("file", file);
+    formData.append("type", "billPdf");
+    formData.append("date", Date);
+    formData.append("boatName", Boat);
+    formData.append("slag", `${Amount}-${Type}`);
```

Comment:
The API input shall be the PDF  `file` you have , and the `pdf-name`
Build the PDF name here,

The API will get a file, a folder and a name and save it. 

The client is responsible to generate the name it wants.